app/scripts/directives/editWebhookTriggers.js

```diff
@@ -0,0 +1,33 @@
+'use strict';
+
+angular.module('openshiftConsole')
+  // Element directive edit BC webhook triggers
+  .directive('editWebhookTriggers', function(ApplicationGenerator) {
+    return {
+      restrict: 'E',
+      scope: {
+        type: "@",
+        typeInfo: "@",
+        triggers: "=",
+        bcName: "@",
+        projectName: "@",
+        form: "="
+      },
+      templateUrl: 'views/directives/edit-webhook-triggers.html',
+      controller: function($scope) {
+        $scope.addWebhookTrigger = function(type) {
+          var webhook = {
+            disabled: false,
+            data: {
+              type: type
+            }
+          };
+          webhook.data[type.toLowerCase()] = {
```

Comment:
I think I'd rather check for known types instead of assuming that `toLowerCase()` will always give the right property name.